admin/admin.go

```diff
@@ -145,6 +160,32 @@ func (m *RPCMethods) GetL1Addresses(args *uint64) *map[string]string {
 		"ProxyAdmin":                        chain.L2Config.L1Addresses.ProxyAdmin.String(),
 		"SuperchainConfig":                  chain.L2Config.L1Addresses.SuperchainConfig.String(),
 	}
-	m.Log.Debug("admin_getL2Addresses")
+	m.log.Debug("admin_getL2Addresses")
 	return &reply
 }
+
+func (m *RPCMethods) GetL2ToL2MessageByMsgHash(args *common.Hash) *interop.L2ToL2Message {
+
+	if args == nil {
+		m.log.Error("valid msg hash not provided", "args", *args)
+		return nil
+	}
+
+	if m.networkConfig.InteropEnabled && m.l2ToL2MsgIndexer != nil {
+
+		storeEntry, err := m.l2ToL2MsgIndexer.Get(*args)
+
+		if err != nil {
+			m.log.Error("failed to get msg", "error", err)
+			return nil
+		}
+
+		reply := storeEntry.Message()
+
+		return reply
```

Comment:
can u verify that this returns correctly (serialized and formatted correctly) when u call the json rpc server? and add a test for it?

Reply:
did to a test on local

![Uploading Screenshot 2024-11-15 at 10.59.09 pm.png…]()
will add test cases too